Counterpart glob pattern

bungeebird opened the pull request:
Here's my current implementation of a counterpart glob pattern. It's a little inelegant, but seems to be working reliably so far! I think it could do with a bit more logging. Would love to hear if you have any notes/comments! Absolutely no rush though! Have a good rest of your weekend! :smiley: 

PS: Whoops, sorry I merged this really weirdly! My bad! Time to get some sleep! :laughing:

KennethWussmann reviewed (changes requested):
Hope you had a great weekend as well!

I think in the long run it might make sense to introduce a business object that wraps around a `File` and `FileSizeTracker`. Now that the counterpart is also associated with it, it could make it easier to pass around state - maybe what you mean by inelegant 😄 But it's perfectly fine like that, maybe if we think about state persistence.

bungeebird commented:
Hey! Sorry for the slow reply! Long week!

I've fixed the order of the checks and added a debug log while waiting for a counterpart file. Good idea re the business object!

I've got a busy weekend coming up, but will try and update whenever I get a chance :smiley:
